gidgethub/actions.py

```diff
@@ -43,3 +42,30 @@ def command(cmd: str, data: str = "", **parameters: str) -> None:
     )
     cmd_parts.append(f"::{data}")
     print("".join(cmd_parts))
+
+
+_DELIMITER = "__GIDGETHUB_DELIMITER__"
+
+
+def setenv(name: str, value: str) -> None:
+    """Create or update an environment variable.
+
+    The change applies to this action and future actions running in the job.
+    """
+    # https://github.com/actions/toolkit/blob/af821474235d3c5e1f49cee7c6cf636abb0874c4/packages/core/src/core.ts#L35-L53
+    os.environ[name] = value
+    # https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#multiline-strings
+    write_value = f"{name}<<{_DELIMITER}{os.linesep}{value}{os.linesep}{_DELIMITER}"
+    with open(os.environ["GITHUB_ENV"], "a", encoding="utf-8") as file:
+        file.write(write_value + os.linesep)
+
+
+def addpath(path: Union[str, pathlib.Path]) -> None:
```

Comment:
What we are really after is the interface, not the concrete implementation we expect practically everyone is using. 😄 

```suggestion
def addpath(path: Union[str, os.PathLike]) -> None:
```